File: Geek Assistant (Android AIO)/Modules/General/Action/GA_SetProgressText.vb
Module GA_SetProgressText


    'Private t As String
    'Private l As Integer

    'Delegate Sub SetLabelTextInvoker(text As String, level As Integer)
    Public Sub Run(text As String, ErrorLevel As Integer)
        My.Settings.LastProgress = text

        Select Case ErrorLevel
            Case -1
                PleaseWait.PleaseWait_ProgressText.ForeColor = SystemColors.ControlText
                Main.ProgressBar.Style = MetroFramework.MetroColorStyle.Green
            Case 0
                PleaseWait.PleaseWait_ProgressText.ForeColor = Color.DarkOrange
                Main.ProgressBar.Style = MetroFramework.MetroColorStyle.Orange
                Media.SystemSounds.Beep.Play()
            Case 1
                PleaseWait.PleaseWait_ProgressText.ForeColor = Color.Firebrick
                PleaseWait.PleaseWait_ProgressText.Text = $"({My.Settings.ErrorCode}) {text}"
                Main.ProgressBar.Style = MetroFramework.MetroColorStyle.Red
                Main.ProgressBarLabel.Text = $"({My.Settings.ErrorCode}) {text}"
                Media.SystemSounds.Asterisk.Play()
                Exit Sub
        End Select
        PleaseWait.PleaseWait_ProgressText.Text = text
        Main.ProgressBarLabel.Text = text

        If My.Settings.VerboseLogging Then
            'LogAppendText(PleaseWait.PleaseWait_ProgressText.Text, 1)
            LogAppendText(Main.ProgressBarLabel.Text, 1)
        End If

        't = text
        'l = level

        ''Dim mi As MethodInvoker = AddressOf Run_Invoked
        ''If Not IsHandleCreated Then
        ''    CreateHandle()
        ''End If
        'If ProgressBarLabel.InvokeRequired Then
        '    ProgressBarLabel.Invoke(New SetLabelTextInvoker(AddressOf ProgressBarLabel_SetText), t, l)
        'Else
        '    ProgressBarLabel_SetText()
        'End If

    End Sub

    'Private Sub ProgressBarLabel_SetText()
    '    Select Case l
    '        Case 0
    '            'PleaseWait.PleaseWait_ProgressText.ForeColor = SystemColors.ControlText
    '            Main.ProgressBarLabel.ForeColor = SystemColors.ControlText
    '        Case 1
    '            'PleaseWait.PleaseWait_ProgressText.ForeColor = Color.DarkOrange
    '            Main.ProgressBarLabel.ForeColor = Color.DarkOrange
    '        Case 2
    '            'PleaseWait.PleaseWait_ProgressText.ForeColor = Color.Firebrick
    '            'PleaseWait.PleaseWait_ProgressText.Text = "(" & My.Settings.ErrorCode & ") " & t
    '            Main.ProgressBarLabel.ForeColor = Color.Firebrick
    '            Main.ProgressBarLabel.Text = "(" & My.Settings.ErrorCode & ") " & t
    '            Exit Sub
    '    End Select
    '    'PleaseWait.PleaseWait_ProgressText.Text = t
    '    Main.ProgressBarLabel.Text = t
    '    If My.Settings.VerbousLogging Then
    '        'LogAppendText(PleaseWait.PleaseWait_ProgressText.Text, 1)
    '        LogAppendText(Main.ProgressBarLabel.Text, 1)
    '    End If
    'End Sub
End Module

File: Geek Assistant (Android AIO)/Modules/General/Action/GA_SetProgressText.cs

internal static partial class GA_SetProgressText {


    // Private t As String
    // Private l As Integer

    // Delegate Sub SetLabelTextInvoker(text As String, level As Integer)
    public static void Run(string text, int ErrorLevel) {
        common.S.LastProgress = text;
        switch (ErrorLevel) {
        case -1: {
            // common.PleaseWait.PleaseWait_ProgressText.ForeColor = SystemColors.ControlText
            common.Home.ProgressBar.Style = MetroFramework.MetroColorStyle.Green;
            break;
        }

        case 0: {
            // common.PleaseWait.PleaseWait_ProgressText.ForeColor = Color.DarkOrange
            common.Home.ProgressBar.Style = MetroFramework.MetroColorStyle.Orange;
            System.Media.SystemSounds.Beep.Play();
            break;
        }

        case 1: {
            // common.PleaseWait.PleaseWait_ProgressText.ForeColor = Color.Firebrick
            // common.PleaseWait.PleaseWait_ProgressText.Text = $"({S.ErrorCode}) {text}"
            common.Home.ProgressBar.Style = MetroFramework.MetroColorStyle.Red;
            common.Home.ProgressBarLabel.Text = $"({common.ErrorInfo.code}) {text}";
            System.Media.SystemSounds.Asterisk.Play();
            return;
        }
        }

        common.PleaseWait.PleaseWait_ProgressText.Text = text;
        common.Home.ProgressBarLabel.Text = text;
        if (common.S.VerboseLogging & ErrorLevel == -1)
            GA_Log.LogAppendText(common.Home.ProgressBarLabel.Text, 1);
    }

    // Private Sub ProgressBarLabel_SetText()
    // Select Case l
    // Case 0
    // 'PleaseWait.PleaseWait_ProgressText.ForeColor = SystemColors.ControlText
    // Main.ProgressBarLabel.ForeColor = SystemColors.ControlText
    // Case 1
    // 'PleaseWait.PleaseWait_ProgressText.ForeColor = Color.DarkOrange
    // Main.ProgressBarLabel.ForeColor = Color.DarkOrange
    // Case 2
    // 'PleaseWait.PleaseWait_ProgressText.ForeColor = Color.Firebrick
    // 'PleaseWait.PleaseWait_ProgressText.Text = "(" & common.S.ErrorCode & ") " & t
    // Main.ProgressBarLabel.ForeColor = Color.Firebrick
    // Main.ProgressBarLabel.Text = "(" & common.S.ErrorCode & ") " & t
    // Exit Sub
    // End Select
    // 'PleaseWait.PleaseWait_ProgressText.Text = t
    // Main.ProgressBarLabel.Text = t
    // If common.S.VerbousLogging Then
    // 'LogAppendText(PleaseWait.PleaseWait_ProgressText.Text, 1)
    // GA_Log.LogAppendText(Main.ProgressBarLabel.Text, 1)
    // End If
    // End Sub
}